File: 0658-find-k-closest-elements/0658-find-k-closest-elements.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        vector<int> answer; 
        sort(arr.begin(), arr.end(), [&](int& a, int& b){
            
            if(abs(a-x) == abs(b-x)){
                return a < b; 
            }

            return abs(a - x) < abs(b - x); 
        });

        for(int i = 0; i < k; i++){
            answer.push_back(arr[i]); 
        }

        sort(answer.begin(),answer.end()); 

        return answer; 
    }
};

File: 0658-find-k-closest-elements/0658-find-k-closest-elements.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        List<Integer> lst = new ArrayList<>(); 
        for(int n : arr) lst.add(n); 

        lst.sort((a,b) -> {
            int resA = Math.abs(a - x); 
            int resB = Math.abs(b - x); 

            if(resA == resB) return Integer.compare(a,b); 

            return Integer.compare(resA,resB); 
        });

        List<Integer> answer = new ArrayList<>(); 
        for(int i = 0; i < k; i++) answer.add(lst.get(i)); 
        answer.sort(Comparator.naturalOrder()); 
        return answer; 
    }
}